### PR TITLE
MRG: Add support for auto_scaling in time_viewer

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR adds support for `auto_scaling` to `_TimeViewer`. It can be enabled by pressing the shortcut `t`. This is still a prototype and I was wondering what would be the best UX actually:

1) Is it more like a 'auto-mode' *(toggle event)*:
* Pressing 'auto-scaling' enables 'auto-mode'
* During playback, the scaling is updated at each step
* If the user modifies the `clim`  values given automatically, they snap back to the detected values

2) Or more like 'help me find a good start' *(action event)*:
* Pressing 'auto-scaling' changes the values
* During playback, the current values are used, no auto-scaling is done
* If the user modifies `clim`, it will be updated accordingly

It's an item of #7162

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
Let's go for 2) for now

**Comment by GuillaumeFavelier** [comment]
What do you think @agramfort , @larsoner ?

**Comment by GuillaumeFavelier** [comment]
I suspect `codecov` had an issue here.

**Comment by larsoner** [comment]
There is a one-sided vs two-sided colormap problem here. Running:
```Python
import os
import mne
from mne.datasets import sample
data_path = sample.data_path()
sample_dir = os.path.join(data_path, 'MEG', 'sample')
subjects_dir = os.path.join(data_path, 'subjects')
fname_stc = os.path.join(sample_dir, 'sample_audvis-meg')
stc = mne.read_source_estimate(fname_stc, subject='sample')
initial_time = 0.1
with mne.viz.use_3d_backend('pyvista'):
    brain = stc.plot(subjects_dir=subjects_dir, initial_time=initial_time,
                     clim=dict(kind='value', pos_lims=[3, 6, 9]),
                     time_viewer=True)
```
I get something correct:

Screenshot from 2020-01-16 09-23-02

Until I hit "t", then I get something that scales in one-sided mode, which is incorrect:

Screenshot from 2020-01-16 09-22-04

Can you fix this, and check that this mode works both in two-sided/divergent mode (`pos_lims`) and one-sided (`lims`) mode?

**Comment by GuillaumeFavelier** [comment]
The user given `colormap` values are now sent to `_limits_to_control_points()` which can detect both in two-sided/divergent mode and one-sided mode.
 
There is also the shortcut `u` to restore to the original `clim`.

I hope this fixed the issues you encountered @larsoner

**Comment by GuillaumeFavelier** [comment]
If possible, I would prefer merge this after https://github.com/mne-tools/mne-python/pull/7219 and resolve the conflicts in here.

**Comment by larsoner** [comment]
Looks like there are no merge conflicts, okay to test and merge @GuillaumeFavelier ?

**Comment by larsoner** [comment]
> The user given colormap values are now sent to _limits_to_control_points() which can detect both two-sided/divergent mode and one-sided mode.

I'm not sure if it should *change* the mode, though. If you initially plot in two-sided mode, it probably shouldn't switch to one-sided (and vice-versa). Currently it seems to, at least for the example where I plot data that are all positive in two-sided mode.

Passing the correct `colormap` in theory should prevent this, since we detect diverging colormaps there.

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -1680,6 +1680,7 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
 
     time_label, times = _handle_time(time_label, time_unit, stc.times)
     # convert control points to locations in colormap
+    user_colormap = colormap  # save the original colormap
```
Comment (larsoner) [inline mne/viz/_3d.py]: I thought that moving this line down *below* the call to `_limits_to_control_points` would fix the colormap-switching behavior, but it didn't (it did the weird positive-only limits for `mne` colormap` behavior). I can live with the current iteration, then

**Comment by larsoner** [comment]
This also seems to work well in testing and I see no merge conflicts so I'll put it in. Thanks @GuillaumeFavelier

**Comment by agramfort** [comment]
amazing !

>

**Pull request merged**